component.js

```diff
@@ -0,0 +1,113 @@
+"use strict";
+
+const $directoryContainer = $(".directory");
+const riseCache = RiseVision.Common.RiseCache;
+const imageCols = [ 2, 4 ];
+
+// Wait for 'WebComponentsReady'.
+      window.addEventListener("WebComponentsReady", () => {
+        const sheet = document.getElementById("googleSheetDirectory");
+
+        // Respond to events it fires.
+          sheet.addEventListener("rise-google-sheet-response", e => {
+              if (e.detail && e.detail.results) {
+
+                displayEvents(e.detail.results);
+              }
+            });
+
+            function displayEvents( events ) {
+              let callsRemaining = 0;
+
+              events.shift();
+
+              // Get total count of calls to be made to Rise Cache.
+              events.forEach( event => {
+                imageCols.forEach( imageCol => {
+                  if ( event[ imageCol ] ) {
+                    callsRemaining++;
+                  }
+                } );
+              } );
+
+              // Set image URLs.
+              events.forEach( event => {
+                imageCols.forEach( imageCol => {
+                  if ( event[ imageCol ] ) {
+                    getFile( event[ imageCol ], url => {
+                      event[ imageCol ] = url;
+                      onFileFetched();
+                    } );
+                  }
+                } );
+              } );
+
+              // Get the proper URL to use from Rise Cache.
+              function getFile( url, callback ) {
+                riseCache.getFile( url, ( response, error ) => {
+                  if ( error ) {
+                    onFileFetched();
+                    return;
+                  }
+
+                  if ( callback ) {
+                    callback( response.url );
+                  }
+                } );
+              }
+
+              function onFileFetched() {
+                callsRemaining--;
+
+                if ( callsRemaining === 0 ) {
+                  appendTheCards( events );
+                }
+              }
+            };
+
+            //append the cards which now have the new urls from the array
+            const appendTheCards = function(finalResults) {
+
+              const eventEntry = finalResults.map( entry => {
+
+                if(entry[2]) {
```

Comment:
@donnapep I have tested the new code structure locally and set up a presentation that I will link to the card. When the appended element may or may not have an image, as in this example, is there a better way to check for this than an` if else` statement with two alternative returns? 

Also at lines 10 and 13 of navigation.js I have added two checks which seem to work for the popup to remove undefined and the broken image links.

Thanks!